### DISCUSSION TITLE
Live stream in external web app

### DISCUSSION THREAD

**Discussion by bartbutenaers** [blakeblackshear/frigate#15706] [Ask A Question] [answered]
Hi everybody,

More and more users in our Node-RED community are integrating Frigate via MQTT.  
Our community is building a new (VueJs) dashboard for Node-RED, and I would love to get some feedback about best practises to show camera live streams in our new dashboard for camera's managed by Frigate.  
In the past we used HLS streams and it looks like go2rtc supports that, but I get the impression that Frigate is more orientated to MSE and WebRTC.  But I might be completely mistaken.
Hopefully somebody can illuminate me: which formats do we need to use, how can we access those streams from Frigate, which libraries could we use in browser, and so on...  

Thanks!!!
Bart

**Comment by hawkeye217** [comment]
Make sure you've set up go2rtc with your Frigate installation, and then just make sure you've got the RTSP restream port open. You can then access those streams from any external app.

https://docs.frigate.video/configuration/restream/#rtsp

**Comment by bartbutenaers** [comment]
@hawkeye217,
Thanks for the (very) quick response!!

Ah ok then I have misinterpreted the concept.  I thought that the live streams (mse, webrtc, ...???) that you produce for the live view (and view of recordings) in the Frigate user interface, would be somehow available also for third party apps.   

Currently I get those RTSP streams directly from my camera's, and I convert them - via ffmpeg - to HLS so that they can be played with the hls.js player in our dashboard (i.e. VueJs dashboard web application).  I thought that such streams (that can be directly played in a browser) would be perhaps somehow available out of the box.  Just wondering because it is not clear to me which kind of streams the Frigate web interface shows, and which player(s) it uses to show the streams...

Perhaps stupid question: is there any advantage for third party apps to get the RTSP streams from Frigate instead of from the camera's directly?  Am I correct that it is better that way because both the cameras cpu usage and the network traffic are a bit relieved that way, by multicasting everything via Go2RTC?

**Reply by hawkeye217** [reply, marked as answer]
Frigate just grabs streams directly from your cameras or from go2rtc and uses frontend js code to play them through the browser in a specific format. Those formats are not available externally in any way. 

https://github.com/blakeblackshear/frigate/tree/dev/web/src/components/player

You'd need to implement a player on your end somehow, grabbing either the camera stream directly or the go2rtc restream. 

The major advantage of go2rtc is that you only have a single connection to your camera.

**Reply by NickM-27** [reply]
To be clear go2rtc can send the stream in many formats including HLS so presumably that'll still work for you. But wherever you're wanting to view the stream, client viewer is required

The easiest to view but also most latency is MP4 stream

**Comment by bartbutenaers** [comment]
Ok thanks guys for illuminating me!!

**Reply by hawkeye217** [reply]
Happy to help!
